Modules/Scripted/DICOMLib/DICOMUtils.py

```diff
@@ -668,3 +668,35 @@ def onNodeAdded(caller, event, calldata):
   slicer.mrmlScene.RemoveObserver(sceneObserverTag)
 
   return loadedNodeIDs
+
+def loadFromDICOMWeb(dicomWebEndpoint, studyUID, seriesUID="", accessToken=""):
+  from dicomweb_client.api import DICOMwebClient
+  client = DICOMwebClient(
+              url = dicomWebEndpoint,
+              headers = { "Authorization": "Bearer {}".format(accessToken) },
+            )
+
+  seriesList = client.search_for_series(study_instance_uid=studyUID)
+  seriesUIDs = []
+  if seriesUID != "":
+    seriesUIDs = [seriesUID]
+  else:
+    for series in seriesList:
+      currentSeriesUID = series['0020000E']['Value'][0]
+      seriesUIDs.append(currentSeriesUID)
+
+  fileNumber = 0
+  for currentSeriesUID in seriesUIDs:
+    instances = client.retrieve_series(
+      study_instance_uid=studyUID,
+      series_instance_uid=currentSeriesUID)
+
+    outputDirectory = (slicer.dicomDatabase.databaseDirectory +
+      "/" + studyUID + "/" + currentSeriesUID + "/")
```

Comment:
IDs may be very long and so when they are used for creating folders then they are often longer than maximum path length on Windows. Instead, you can create unique folder using date&time and/or hashes.